lxd/storage/drivers/driver_pure_util.go

```diff
@@ -517,6 +572,475 @@ func (p *pureClient) disconnectHostFromVolume(poolName string, volName string, h
+		target, err := d.client().getTarget()
+		if err != nil {
+			return err
+		}
+
+		// Disconnect from the target.
+		err = d.disconnect(*target)
+		if err != nil {
+			return err
+		}
+
+		// Remove the host from PureStorage.
+		err = d.client().deleteHost(host.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getMappedDevPath returns the local device path for the given volume.
+// Indicate with mapVolume if the volume should get mapped to the system if it isn't present.
+func (d *pure) getMappedDevPath(vol Volume, mapVolume bool) (string, revert.Hook, error) {
+	revert := revert.New()
+	defer revert.Fail()
+
+	if mapVolume {
+		err := d.mapVolume(vol)
+		if err != nil {
+			return "", nil, err
+		}
+
+		revert.Add(func() { _ = d.unmapVolume(vol) })
+	}
+
+	// findDevPathFunc has to be called in a loop with a set timeout to ensure
+	// all the necessary directories and devices can be discovered.
+	findDevPathFunc := func(diskPrefix string, diskSuffix string) (string, error) {
+		var diskPaths []string
+
+		// If there are no other disks on the system by id, the directory might not even be there.
+		// Returns ENOENT in case the by-id/ directory does not exist.
+		diskPaths, err := resources.GetDisksByID(diskPrefix)
+		if err != nil {
+			return "", err
+		}
+
+		for _, diskPath := range diskPaths {
+			// Skip the disk if it is only a partition of the actual volume.
+			if strings.Contains(diskPath, "-part") {
+				continue
+			}
+
+			// Skip volumes that do not have volume's pool and name suffix.
+			if !strings.HasSuffix(diskPath, diskSuffix) {
+				continue
+			}
+
+			// The actual device might not already be created.
+			// Returns ENOENT in case the device does not exist.
+			devPath, err := filepath.EvalSymlinks(diskPath)
+			if err != nil {
+				return "", err
+			}
+
+			return devPath, nil
+		}
+
+		return "", nil
+	}
+
+	var volumeDevPath string
+	var diskPrefix string
+	var diskSuffix string
+
+	volName, err := d.getVolumeName(vol)
+	if err != nil {
+		return "", nil, err
+	}
+
+	switch d.config["pure.mode"] {
+	case pureModeISCSI:
+		diskPrefix = "scsi-"
+		diskSuffix = fmt.Sprintf("%s::%s", vol.pool, volName)
+	}
+
+	timeout := time.Now().Add(15 * time.Second)
+
+	// It might take a while to create the local disk.
+	// Retry until it can be found.
+	for {
+		if time.Now().After(timeout) {
+			return "", nil, fmt.Errorf("Failed to locate device for volume %q: Timeout exceeded", vol.name)
+		}
+
+		devPath, err := findDevPathFunc(diskPrefix, diskSuffix)
+		if err != nil {
+			// Try again if on of the directories cannot be found.
+			if errors.Is(err, unix.ENOENT) {
+				continue
+			}
+
+			return "", nil, err
+		}
+
+		if devPath != "" {
+			volumeDevPath = devPath
+			break
+		}
+
+		// Exit if the volume wasn't explicitly mapped.
+		// Doing a retry would run into the timeout when the device isn't mapped.
+		if !mapVolume {
+			break
+		}
+
+		time.Sleep(10 * time.Millisecond)
```

Comment:
this seems too short, and will likely cause this to be repeated many times in a tight loop causing CPU spikes.

Other sleeps in the storage package tend to me in the hundreds of milliseconds, for example `waitGone` uses `time.Sleep(500 * time.Millisecond)`

Reply:
powerflex likely needs to be updated here too.